avogadro/io/poscarformat.cpp

```diff
@@ -0,0 +1,337 @@
+    cellMat(1,i) = lexicalCast<double>(stringSplit.at(1)) * scalingFactor;
+    cellMat(2,i) = lexicalCast<double>(stringSplit.at(2)) * scalingFactor;
+  }
+
+  // Sometimes, atomic symbols go here.
+  getline(inStream, line);
+  stringSplit = split(line, /*delimiter=*/ ' ');
+
+  if (stringSplit.empty()) {
+    appendError("Error reading numbers of atom types in POSCAR");
+    return false;
+  }
+
+  // Try a lexical cast here. If it fails, assume we have an atomic symbols list
+  lexicalCast<uint>(trimmed(stringSplit.at(0)), ok);
+  vector<string> symbolsList;
+  vector<unsigned char> atomicNumbers;
+
+  if (!ok) {
+    // Assume atomic symbols are here and store them
+    symbolsList = split(line, /*delimiter=*/ ' ');
+    // Store atomic nums
+    for (size_t i = 0; i < symbolsList.size(); ++i)
+      atomicNumbers.push_back(Elements::atomicNumberFromSymbol(symbolsList.at(i)));
+    // This next one should be atom types
+    getline(inStream, line);
+  }
+  // If the atomic symbols aren't here, try to find them in the title
+  // In Vasp 4.x, symbols are in the title like so: " O4H2 <restOfTitle>"
+  else {
+    stringSplit = split(title, /*delimiter=*/ ' ');
+    if (stringSplit.size() != 0) {
+      string trimmedFormula = trimmed(stringSplit.at(0));
+      // Let's replace all numbers with spaces
+      for (size_t i = 0; i < trimmedFormula.size(); ++i) {
+        if (isdigit(trimmedFormula.at(i)))
+          trimmedFormula[i] = ' ';
+      }
+      // Now get the symbols with a simple space split
+      symbolsList = split(trimmedFormula, /*delimiter=*/ ' ');
+      for (size_t i = 0; i < symbolsList.size(); ++i)
+        atomicNumbers.push_back(Elements::atomicNumberFromSymbol(symbolsList.at(i)));
+    }
+  }
+
+  stringSplit = split(line, /*delimiter=*/ ' ');
+  vector<uint> atomCounts;
+  for (size_t i = 0; i < stringSplit.size(); ++i) {
+    uint atomCount = lexicalCast<uint>(stringSplit.at(i));
+    atomCounts.push_back(atomCount);
+  }
+
+  // If we never filled up the atomic numbers, fill them up
+  // now with "1, 2, 3..."
+  if (atomicNumbers.size() == 0)
+    for (size_t i = 1; i <= atomCounts.size(); ++i) atomicNumbers.push_back(i);
+
+  if (atomicNumbers.size() != atomCounts.size()) {
+    appendError("Error: numSymbols and numTypes are not equal in POSCAR!");
+    return false;
+  }
+
+  // Starts with either [Ss]elective dynamics, [KkCc]artesian, or
+  // other for fractional coords.
+  getline(inStream, line);
+  line = trimmed(line);
+
+  // If selective dynamics, get the next line
+  if (line.empty() || line.at(0) == 'S' || line.at(0) == 's')
+    getline(inStream, line);
+
+  line = trimmed(line);
+  if (line.empty()) {
+    appendError("Error determining Direct or Cartesian in POSCAR");
+    return false;
+  }
+
+  bool cart;
+  // Check if we're using cartesian or fractional coordinates:
+  if (line.at(0) == 'K' || line.at(0) == 'k' ||
+      line.at(0) == 'C' || line.at(0) == 'c' )
```

Comment:
When the if statement takes more than one line we prefer to use { } to make it clear where the statement is.